bundles/org.openhab.binding.modbus.studer/src/main/java/org/openhab/binding/modbus/studer/internal/StuderParser.java

```diff
@@ -0,0 +1,237 @@
+            default:
+                return VTType.UNKNOWN;
+        }
+    }
+
+    public enum VTMode {
+        NIGHT(0),
+        STARTUP(1),
+        CHARGER(3),
+        SECURITY(5),
+        OFF(6),
+        CHARGE(8),
+        CHARGEV(9),
+        CHARGEI(10),
+        CHARGET(11),
+        CHIBSP(12),
+        UNKNOWN(-1);
+
+        private final int code;
+
+        VTMode(int code) {
+            this.code = code;
+        }
+
+        public int code() {
+            return this.code;
+        }
+    }
+
+    public static VTMode getVTModeByCode(int code) {
+        switch (code) {
+            case 0:
+                return VTMode.NIGHT;
+            case 1:
+                return VTMode.STARTUP;
+            case 3:
+                return VTMode.CHARGER;
+            case 5:
+                return VTMode.SECURITY;
+            case 6:
+                return VTMode.OFF;
+            case 8:
+                return VTMode.CHARGE;
+            case 9:
+                return VTMode.CHARGEV;
+            case 10:
+                return VTMode.CHARGEI;
+            case 11:
+                return VTMode.CHARGET;
+            case 12:
+                return VTMode.CHIBSP;
+            default:
+                return VTMode.UNKNOWN;
+        }
+    }
+
+    public enum VSMode {
+        NIGHT(0),
+        SECURITY(1),
+        OFF(2),
+        CHARGE(3),
+        CHARGEV(4),
+        CHARGEI(5),
+        CHARGEP(6),
+        CHARGEIPV(7),
+        CHARGET(8),
+        CHIBSP(10),
+        UNKNOWN(-1);
+
+        private final int code;
+
+        VSMode(int code) {
+            this.code = code;
+        }
+
+        public int code() {
+            return this.code;
+        }
+    }
+
+    public static VSMode getVSModeByCode(int code) {
+        switch (code) {
+            case 0:
+                return VSMode.NIGHT;
+            case 1:
+                return VSMode.SECURITY;
+            case 2:
+                return VSMode.OFF;
+            case 3:
+                return VSMode.CHARGE;
+            case 4:
+                return VSMode.CHARGEV;
+            case 5:
+                return VSMode.CHARGEI;
+            case 6:
+                return VSMode.CHARGEP;
+            case 7:
+                return VSMode.CHARGEIPV;
+            case 8:
+                return VSMode.CHARGET;
+            case 10:
+                return VSMode.CHIBSP;
+            default:
+                return VSMode.UNKNOWN;
+        }
+    }
+
+    /**
+     * Convert an hex string to float
+     *
+     * @param hex string to convert from
+     * @return the converted float
+     */
+    public @Nullable Object hexToFloat(String hex) {
+        String t = hex.replaceAll(" ", "");
+        try {
+            float f = Float.intBitsToFloat((int) Long.parseLong(t, 16));
+            if (Float.isNaN(f)) {
+                return null;
+            } else {
+                return f;
+            }
+        } catch (Exception e) {
```

Comment:
You should specify the concrete exception you want to catch, since catching `Exception` catches also all runtime exceptions.